src/v2/kuadrant/pipeline/tasks/mod.rs

```diff
@@ -14,9 +16,9 @@ pub use token_usage::TokenUsageTask;
 
 use crate::v2::kuadrant::ReqRespCtx;
 
-pub type ResponseProcessor = dyn FnOnce(&mut ReqRespCtx, u32, usize) -> TaskOutcome;
+//todo(refactor): this now has the signature of a task; should it be one?
+pub type ResponseProcessor = dyn FnOnce(&mut ReqRespCtx) -> TaskOutcome;
```

Comment:
This is now a function closure that looks exactly like a `task.apply(ctx)`.. maybe it should just be building a task